components/tenant-mgt/org.wso2.carbon.tenant.mgt/src/main/java/org/wso2/carbon/tenant/mgt/util/TenantIDNDataDeletionUtil.java

```diff
@@ -0,0 +1,100 @@
+    private static final String DELETE_IDP = "DELETE FROM IDP WHERE TENANT_ID = ?";
+    private static final String DELETE_IDP_AUTHENTICATOR = "DELETE FROM IDP_AUTHENTICATOR WHERE TENANT_ID = ?";
+    private static final String DELETE_IDP_AUTHENTICATOR_PROPERTY = "DELETE FROM IDP_AUTHENTICATOR_PROPERTY WHERE TENANT_ID = ?";
+    private static final String DELETE_IDP_METADATA = "DELETE FROM IDP_METADATA WHERE TENANT_ID = ?";
+    private static final String DELETE_IDP_PROVISIONING_CONFIG = "DELETE FROM IDP_PROVISIONING_CONFIG WHERE TENANT_ID = ?";
+    private static final String DELETE_IDP_PROV_CONFIG_PROPERTY = "DELETE FROM IDP_PROV_CONFIG_PROPERTY WHERE TENANT_ID = ?";
+
+    /**
+     * Delete all tenant information related to tenant stored in IDN tables
+     *
+     * @param tenantId id of tenant whose data should be deleted
+     * @param conn     database connection object
+     * @throws SQLException thrown if an error occurs while executing the queries
+     */
+    protected static void deleteTenantIDNData(int tenantId, Connection conn) throws SQLException {
+
+        try {
+            conn.setAutoCommit(false);
+            executeDeleteQuery(conn, DELETE_CM_PURPOSE_CATEGORY, tenantId);
+            executeDeleteQuery(conn, DELETE_IDN_CLAIM, tenantId);
+            executeDeleteQuery(conn, DELETE_IDN_CLAIM_DIALECT, tenantId);
+            executeDeleteQuery(conn, DELETE_IDN_CLAIM_MAPPED_ATTRIBUTE, tenantId);
+            executeDeleteQuery(conn, DELETE_IDN_CLAIM_MAPPING, tenantId);
+            executeDeleteQuery(conn, DELETE_IDN_CLAIM_PROPERTY, tenantId);
+            executeDeleteQuery(conn, DELETE_IDN_OIDC_SCOPE, tenantId);
+            executeDeleteQuery(conn, DELETE_IDP, tenantId);
+            executeDeleteQuery(conn, DELETE_IDP_AUTHENTICATOR, tenantId);
+            executeDeleteQuery(conn, DELETE_IDP_AUTHENTICATOR_PROPERTY, tenantId);
+            executeDeleteQuery(conn, DELETE_IDP_METADATA, tenantId);
+            executeDeleteQuery(conn, DELETE_IDP_PROVISIONING_CONFIG, tenantId);
+            executeDeleteQuery(conn, DELETE_IDP_PROV_CONFIG_PROPERTY, tenantId);
+            conn.commit();
+        } catch (Exception e) {
+            conn.rollback();
+            String errorMsg = "An error occurred while deleting identity data for tenant: " + tenantId;
+            log.error(errorMsg, e);
+            throw new SQLException(errorMsg, e);
+        } finally {
+            conn.close();
+        }
+    }
+
+    /**
+     * Initialise prepared statements for given query and execute the prepared statement.
+     *
+     * @param conn     database connection object
+     * @param query    query for prepared statement
+     * @param tenantId tenant id
+     * @throws SQLException thrown if an error occurs while executing the query.
+     */
+    private static void executeDeleteQuery(Connection conn, String query, int tenantId)
+            throws SQLException {
+
```

Comment:
Remove unnecessary new line at the beginning of each method.

Reply:
Removed